src/app/calendar-app/calendar.service.ts

```diff
@@ -282,4 +548,40 @@ export class CalendarService implements OnDestroy {
             this.activities.end(Activity.RefreshingCalendars);
         });
     }
+
+    // check if we have enough events to show another month worth
+    // (forward and back), generate if not
+    updateEventList(viewPeriod?: ViewPeriod): Promise<void> {
```

Comment:
Hmm, any reason why this returns a Promise? Seems completely synchronous to me.

Reply:
Erm, dunno? cargo-culting probably!